go/porcelain/http/http.go

```diff
@@ -31,10 +31,11 @@ func NewRetryableTransport(tr runtime.ClientTransport, attempts int) *RetryableT
 }
 
 func (t *RetryableTransport) Submit(op *runtime.ClientOperation) (interface{}, error) {
-	client := op.Client
-
-	if client == nil {
-		client = http.DefaultClient
+	client := &http.Client{}
+	if op.Client == nil {
+		*client = *http.DefaultClient
+	} else {
+		*client = *op.Client
```

Comment:
why not just use a local copy? (then turn it into a reference when setting on the op)
```suggestion
	var client http.Client
	if op.Client == nil {
		client = *http.DefaultClient
	} else {
		client = *op.Client
```

Reply:
Sure, that works too. `client = *http.DefaultClient` doesn't really do anything (in either version) because the default client is `&http.Client{}` but I think it's bit more explicit.